R/auto.R

```diff
@@ -57,68 +77,108 @@ guess_next_impl <- function(version) {
   return(which)
 }
 
-pre_release_impl <- function(which, force) {
-  # FIXME: Needs repair in create_release_branch()
-  stopifnot(!force)
+init_release_impl <- function(which, force) {
+  # FIXME: Slightly different process for releasing from the non-main branch
+  # - Don't bump
+  # - Use HEAD instead of HEAD^ for creating the release branch
 
-  # https://github.com/r-lib/gert/issues/139
-  stopifnot(gert::git_branch() != "HEAD")
+  # Checking if it's an orphan branch: https://github.com/r-lib/gert/issues/139
+  stopifnot(get_branch_name() != "HEAD")
 
   # check PAT scopes for PR for early abort
   check_gh_pat("repo")
 
+  fledgeling <- read_fledgling()
+
+  new_version <- fledge_guess_version(fledgeling$version, which)
+
+  if (!force) {
+    check_release_branch(new_version)
+  }
+
   cat(boxx("pre-release", border_style = "double"))
 
   # Begin extension points
   # End extension points
 
-  # We expect that this branch is pushed already, ok to fail here
-  main_branch <- get_branch_name()
-  remote_name <- get_remote_name(main_branch)
-
-  cli_h1("1. Creating a release branch and getting ready")
+  cli_h1("1. Wrapping up development")
 
-  # bump version on main branch to version set by user
-  # Avoid `bump_version()` to avoid showing `NEWS.md` at this stage,
-  # because it changes as we jump between branches.
-  update_news(which = which)
-  commit_version()
-
-  # switch to release branch and update cran-comments
-  release_branch <- create_release_branch(force)
-  switch_branch(release_branch)
-  update_cran_comments()
+  # Bump dev if needed, bail out if bumped
+  new_fledgeling <- bump_version_impl(fledgeling, "dev", no_change_behavior = "noop")
+  if (!identical(new_fledgeling, fledgeling)) {
+    return()
+  }
+  rm(new_fledgeling)
```

Comment:
for some reason this surprises me?

Reply:
Good point, we can extract a function here.

Reply:
When we're touching this code the next time 🤪